crates/swc_ecma_utils/src/expr_ignore_span.rs

```diff
@@ -0,0 +1,132 @@
+use std::{borrow::Cow, fmt::Debug, hash::Hash};
+
+use rustc_hash::FxHashSet;
+use swc_common::{util::take::Take, EqIgnoreSpan, Mark, DUMMY_SP};
+use swc_ecma_ast::{Expr, Ident, MemberProp};
+
+use crate::{member_expr, quote_expr};
+
+pub struct NodeIgnoringSpan<'a, Node: ToOwned + Debug>(Cow<'a, Node>);
+
+impl<'a, Node: ToOwned + Debug> Debug for NodeIgnoringSpan<'a, Node> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("NodeIgnoringSpan").field(&*self.0).finish()
+    }
+}
+
+impl<'a, Node: ToOwned + Debug> NodeIgnoringSpan<'a, Node> {
+    #[inline]
+    pub fn borrowed(expr: &'a Node) -> Self {
+        Self(Cow::Borrowed(expr))
+    }
+
+    #[inline]
+    pub fn owned(expr: <Node as ToOwned>::Owned) -> Self {
+        Self(Cow::Owned(expr))
+    }
+}
+
+impl<'a, Node: EqIgnoreSpan + ToOwned + Debug> PartialEq for NodeIgnoringSpan<'a, Node> {
+    fn eq(&self, other: &Self) -> bool {
+        Ident::within_ignored_ctxt(|| self.0.eq_ignore_span(&other.0))
+    }
+}
+
+impl<'a, Node: EqIgnoreSpan + ToOwned + Debug> Eq for NodeIgnoringSpan<'a, Node> {}
+
+// TODO: This only a workaround for Expr. we need something like
+// `hash_ignore_span` for each node in the end.
```

Comment:
I think we need `hash_ignore_span` in the end if we want to make this implementation energetic.

And the code could be rewritten like
```rust
impl<'a, Node: HashIgnoreSpan> Hash for NodeIgnoringSpan<'a, Node> {
    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
          self.hash_ignore_span(state)
        }
    }
}
```

However, I think this could be done in another PR.